### ISSUE TITLE
Is there a way to useRef inside the component string ?

### ISSUE THREAD

**Issue by Search-For-a-Friend** [dev-javascript/string-to-react-component#2]
Hello and thank you for this much useful tool,

I am using your StringToReactComponent in order to render a 3D Model from a .jsx file generated with [gltfjsx](https://github.com/pmndrs/gltfjsx)
Here is a sample of the adapted .jsx file I read from backend and display through the StringToReactComponent :

`(props) => { 

  const {nodes, materials, animations, useAnimations, useEffect, ...rest} = props;
  
  const {useRef}=React;

  const ref = useRef();
  const actions = useAnimations(animations, ref);

  return (
    <group ref={ref} {...props}>
       (rest of the code of the 3D Model from the jsx file)
    </group>
  )
}`

When I try to display this, I get the error :

"TypeError: Assignment to constant variable."
If I remove the "ref={ref}" the component displays everything perfectly, without error.

Maybe your component is not thought for this kind of usecase though ?

**Comment by dev-javascript** [comment]
Hello thanks for asking

There is not any difference between components which are created by this library and other components which you create them statically

I think this issue is not related to this plugin, the below code works correctly :
```
`(props) => {
  const { useRef, useLayoutEffect } = React;
  const ref = useRef();

  useLayoutEffect(() => {
    ref.current.style.background = 'red';
  }, [ref]);

  return (<div ref={ref} style={{ padding: '10px' }} />);
}`
```

Could you create a codesandbox for this issue?

**Comment by Search-For-a-Friend** [comment]
Okay thank you for answering ! Let me try again, then I come back to you with a sandbox if something is still off.
Thanks again for your time.

**Comment by Search-For-a-Friend** [comment]
Okay so here is my sand box link :
https://codesandbox.io/s/elastic-mcclintock-fn4ps9?file=/src/viewer.js
If you remove the `ref={ref}` in public/model.jsx everything works perfectly.

Thanks again for your time, the issue is probably something I missed.

**Comment by dev-javascript** [comment]
You're most welcome.
The answer is :

```
`
(props) => {
  const { nodes: nodes, materials: materials, ...rest } = props;
  const { useRef } = React;
  const ref = useRef();
  const all_props = { ...props };
  all_props.ref = ref;
  return (
    <group  {...all_props} >
(rest of the code of the 3D Model from the jsx file)

)
}`  
```    
The above code works correctly without error. 
This issue is related to `@react-three/fiber` package, I have not worked with `@react-three/fiber` package, but it seems that just **one object** as props can be passed to the `group` element.


Examples :
```
`
(props) => {
  const { nodes: nodes, materials: materials, ...rest } = props;
  return (
    <group  {...props} ref={null} > {/* We get the "TypeError: Assignment to constant variable" error here */}
     (rest of the code of the 3D Model from the jsx file)
  )
}`  
```
or

```
`
(props) => {
  const { nodes: nodes, materials: materials, ...rest } = props;
  return (
    <group  {...props} className="XXX" >  {/* We get the "TypeError: Assignment to constant variable" error here */}
     (rest of the code of the 3D Model from the jsx file)
  )
}`  
```
```
`
(props) => {
  const { nodes: nodes, materials: materials, ...rest } = props;
  const all_props = { ...props };
  all_props.className = "XXX";
  return (
    <group  {...all_props } >  {/* works correctly */}
     (rest of the code of the 3D Model from the jsx file)
  )
}`  
```

**Comment by Search-For-a-Friend** [comment]
Ok thank you so much for your time and for working on a package you are not used to. I will test this asap

PS : it worked perfectly, thank you very much

**Comment by dev-javascript** [comment]
You're most welcome
